locations/spiders/brewdog.py

```diff
@@ -1,53 +1,40 @@
-import re
-import urllib.parse
+import json
 
 import scrapy
 
 from locations.hours import OpeningHours
 from locations.items import GeojsonPointItem
 
-kHoursRe = re.compile(r"(?P<day>\w+):\s+(?P<open_time>\S+) - (?P<close_time>\S+)")
-
 
 class BrewdogSpider(scrapy.Spider):
     name = "brewdog"
     item_attributes = {"brand": "BrewDog", "brand_wikidata": "Q911367"}
     allowed_domains = ["www.brewdog.com"]
-    start_urls = (
-        "https://www.brewdog.com/uk/bar_pages/bar/locator/view_all/1/filter/uk/",
-        "https://www.brewdog.com/usa/bar_pages/bar/locator/view_all/1/filter/usa/",
-        "https://www.brewdog.com/eu_en/bar_pages/bar/locator/view_all/1/filter/global/",
-    )
+    start_urls = ["https://www.brewdog.com/uk/bar-locator#ALL"]
 
     def parse(self, response):
-        yield from response.follow_all(css="a.bar-listing__results__bar__buttons__view", callback=self.parse_bar)
-
-    def parse_bar(self, response):
-        name = response.xpath('//meta[@name="title"]/@content').extract_first()
-        addr = next(s for s in response.xpath('//div[@id="contact"]//text()').extract() if s.strip())
-
-        opening_hours = OpeningHours()
-        for row in response.xpath('//div[@id="opening-hours"]/p/text()').extract()[:7]:
-            groups = kHoursRe.search(row)
-            if groups is None:
-                continue
-            day = groups["day"][:2]
-            opening_hours.add_range(day, groups["open_time"], groups["close_time"], "%I:%M%p")
-
-        map_url = response.xpath('//iframe[@class="bar-detail__map"]/@src').extract_first()
-        if map_url:
-            query = urllib.parse.urlparse(map_url).query
-            [q] = urllib.parse.parse_qs(query)["q"]
-            lat, lon = q.split(",")
-        else:
-            lat = lon = None
-
-        return GeojsonPointItem(
-            ref=response.url,
-            website=response.url,
-            name=name,
-            addr_full=addr,
-            lat=lat,
-            lon=lon,
-            opening_hours=opening_hours.as_opening_hours(),
-        )
+        data = response.xpath('//script[@id="__NEXT_DATA__"]/text()').get()
+        data_json = json.loads(data)
+        bars = data_json["props"]["pageProps"]["content"]
+        item = GeojsonPointItem()
+        for bar in bars:
+            item["ref"] = bar["sys"].get("id")
+            item["name"] = bar["fields"].get("name")
+            item["addr_full"] = bar["fields"].get("address")
+            item["phone"] = bar["fields"].get("contactNumber")
+            item["email"] = bar["fields"].get("contactEmail")
+            item["lat"] = bar["fields"].get("location", {}).get("lat")
+            item["lon"] = bar["fields"].get("location", {}).get("lon")
```

Comment:
Since the current code has been written this isn't important, but for future we have `DictParser.parse` that can do most of the common fields for us automatically, I think it can do all of these.

```suggestion
            item = DictParser.parse(bar["fields"])
            item["ref"] = bar["sys"]["id"]
```